lib/Parser/JSParserImpl.cpp

```diff
@@ -20,6 +20,27 @@ namespace hermes {
 namespace parser {
 namespace detail {
 
+namespace {
+
+bool canNextTokenFollowModifier(OptValue<TokenKind> optNext) {
```

Comment:
Can we have a doc-comment on this function? It needs a description of what it does, return value, etc. Also, please put a `TS` in this function's name to indicate it's only for typescript.

Reply:
That's fair. Then perhaps I might move it to `class JSParserImpl` as a TS-specific method so we can have the implementation in `JSParserImpl-ts.cpp` file. Does that make sense to you?

Similarly, do you think `getAccessibilityIdent()` that I added should also be TS-specific? If so, I will do the same.

Reply:
> getAccessibilityIdent()

It seems like instead of storing a `TokenKind` you can just store `NodeLabel accessibility` and do `accessibility = tok_->getResWordIdent()` at the assignment points, to avoid needing that kind of function any more. Is there something that prevents using `getResWordIdent` and `getResWordOrIdent` for problems like these in this context?

Reply:
No, nothing particular (other than my ignorance of the API 😀). I think that would be better. Thank you for the suggestion!